### ISSUE TITLE
这项工作很感兴趣，请问一块3090能跑这个实验吗

### ISSUE THREAD

**Issue by zhangyubo1021** [whu-pzhang/ASANet#2]

**Comment by whu-pzhang** [comment]
可以，我们实验是在2块RTX8000上跑的

**Comment by zhangyubo1021** [comment]
> 可以，我们实验是在2块RTX8000上跑的

请问需要跑大概多长时间

**Comment by cccty121** [comment]
我用h800跑asanet-whuoptasr四个小时 用3060ti跑asanet-whuoptasr大概26个小时

**Comment by Seven1099** [comment]
> 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时

你好请问ddrh和whu数据集下载后需要处理吗，谢谢你

**Comment by zhangyubo1021** [comment]
> 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
你好请问ddrh和whu数据集 是怎么划分的

**Comment by zhangyubo1021** [comment]
> 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时

你好 请问能告诉我如何划分whu数据集吗

**Comment by cccty121** [comment]
> > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> 
> 你好 请问能告诉我如何划分whu数据集吗

import os
import random
import numpy as np

# 设置随机种子，以确保划分一致
random_seed = 42
random.seed(random_seed)

# 输入文件夹
root_folder = "
img_folder = root_folder + 'lbl'
# 获取文件夹中的所有文件
files = os.listdir(img_folder)

# 确保文件按顺序排序
files.sort()

# 划分比例
train_ratio = 0.8
split_index = int(len(files) * train_ratio)

# 划分数据集
train_files = files[:split_index]
val_files = files[split_index:]

# 输出文件路径
train_txt = os.path.join(root_folder, 'train_list.txt')
val_txt = os.path.join(root_folder, 'val_list.txt')

# 写入训练集文件名到train.txt
with open(train_txt, 'w') as train_file:
    for filename in train_files:
        train_file.write(filename + '\n')

# 写入验证集文件名到validation.txt
with open(val_txt, 'w') as val_file:
    for filename in val_files:
        val_file.write(filename + '\n')

print(f"数据集划分完成，并保存到 {train_txt} 和 {val_txt}")

**Comment by cccty121** [comment]
> > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> 
> 你好 请问能告诉我如何划分whu数据集吗

用tools/split.py路径下的工具进行 8：2划分

**Comment by cccty121** [comment]
> > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> 
> 你好请问ddrh和whu数据集下载后需要处理吗，谢谢你

需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py    tools/split.py   这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中

**Comment by zhangyubo1021** [comment]
> > > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> > 
> > 
> > 您好请问ddrh和whu数据集下载后需要处理吗，谢谢
> 
> 需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py tools/split.py 这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中

谢谢你

**Comment by zhangyubo1021** [comment]
> > > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> > 
> > 
> > 您好请问ddrh和whu数据集下载后需要处理吗，谢谢
> 
> 需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py tools/split.py 这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中

请问您说的将标签转为2d形式输入到mmseg库中具体应该怎么操作，您可以详细描述一下吗

**Comment by cccty121** [comment]
> > > > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> > > 
> > > 
> > > 您好请问ddrh和whu数据集下载后需要处理吗，谢谢
> > 
> > 
> > 需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py tools/split.py 这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中
> 
> 请问您说的将标签转为2d形式输入到mmseg库中具体应该怎么操作，您可以详细描述一下吗

具体可以参考mmseg官方流程，如果还有疑问，我可以给你一份我做的配置文档，不过你不是这个项目的作者之一吗?

**Comment by cccty121** [comment]
> > > > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> > > 
> > > 
> > > 您好请问ddrh和whu数据集下载后需要处理吗，谢谢
> > 
> > 
> > 需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py tools/split.py 这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中
> 
> 请问您说的将标签转为2d形式输入到mmseg库中具体应该怎么操作，您可以详细描述一下吗

具体步骤可以参考mmseg官方文档，你运行perpare_whu-opt-sar.py文件之后，把处理好的图像的路径输入到configs/_base_/datasets/whu-opt-sar-RGBS_512x512.py，再运行python train.py /home/ht/workspace/nuaa/asanet/ASANet-main/configs/ASANet/ASANet_convnextv2-tiny_2xb4-amp-80k_whu-opt-sar-RGBS_512x512.py就可，如果只有一张显卡，记得改为norm_cfg = dict(type='**BN**', requires_grad=True)，如果你是ssh链接的服务器，建议将wandb改为离线模式：
vis_backends = [
    dict(type='LocalVisBackend'),
    dict(type='WandbVisBackend',
         init_kwargs=dict(
             project='PIERGBSFusion',#项目名称 随便改
             name='upernet-convnextv2-tiny_SFM_CFM',#模型名字
             **mode='offline'  # 启用离线模式**
         ))
如果你还不能复现，我这有做好的配置流程。

**Comment by zhangyubo1021** [comment]
> > > > > 我用h800跑asanet-whuoptasr四个小时用3060ti跑asanet-whuoptasr大概26个小时
> > > > 
> > > > 
> > > > 您好请问ddrh和whu数据集下载后需要处理吗，谢谢
> > > 
> > > 
> > > 需要处理哦，处理流程请看tools/perpare_whu-opt-sar.py tools/split.py 这两个文件，因为whu的标签是0-10像素分类，记得将标签转为2d形式输入到mmseg库中
> > 
> > 
> > 请问您说的将标签转为2d形式输入到mmseg库中具体应该怎么操作，您可以详细描述一下吗
> 
> 具体步骤可以参考mmseg官方文档，你运行perpare_whu-opt-sar.py文件之后，把处理好的图像的路径输入到configs/ _base_ /datasets/whu-opt-sar-RGBS_512x512.py，再运行python train。 py /home/ht/workspace/nuaa/asanet/ASANet-main/configs/ASANet/ASANet_convnextv2-tiny_2xb4-amp-80k_whu-opt-sar-RGBS_512x512.py就可以，如果只有一张显卡，记得一定norm_cfg = dict(类型=' **BN** ', require_grad=True)，如果你是ssh链接的服务器，建议将wandb改为离线模式： vis_backends = [ dict(type='LocalVisBackend'), dict(type='WandbVisBackend', init_kwargs=dict( project='PIERGBSFusion',#项目名称便捷改 name='upernet-convnextv2-tiny_SFM_CFM',#模型名称 **mode='offline' # 启用离线模式** )) 如果你还不能恢复现，我这有完整的配置流程。

谢谢你 我跑出来了 感谢你的耐心解答
